backbone.js

```diff
@@ -890,15 +891,19 @@
       // Silently sort the collection if appropriate.
       if (sort) this.sort({silent: true});
 
-      // Unless silenced, it's time to fire all appropriate add/sort events.
+      // Unless silenced, it's time to fire all appropriate add/update/sort events.
       if (!options.silent) {
         var addOpts = at != null ? _.clone(options) : options;
         for (var i = 0; i < toAdd.length; i++) {
           if (at != null) addOpts.index = at + i;
           (model = toAdd[i]).trigger('add', model, this, addOpts);
         }
+        if (toAdd.length || toRemove.length) {
+          if (toAdd.length) options.addedModels = toAdd;
+          if (toRemove.length) options.previousModels = toRemove;
```

Comment:
The only other `previousModels` always attaches. I think attaching an empty array here is fine as well.

Also, can we rename this `removedModels`? `previousModels` makes sense in `#reset` where every model is "previous", but here only a subset are being removed.